packages/dcos-launch/extra/launch/config.py

```diff
@@ -1,16 +1,38 @@
+import atexit
 import copy
 import os
 
 import yaml
 
-import dcos_installer.config
 import gen
 import launch.util
 import ssh.validate
 import test_util.aws
 from gen.internals import resolve_configuration, Scope, Source, Target, validate_one_of
 from pkgpanda.util import load_string, load_yaml, YamlParseError
 
+# dcos_installer.config will directly import gen.build_deploy.util
+# which expects to find the image commit in the environment or in
+# a directory-local git tree **at import time**. Therefore, the
+# environment variable DCOS_IMAGE_COMMIT must be set here.
+if 'DCOS_IMAGE_COMMIT' not in os.environ:
+    os.environ['DCOS_IMAGE_COMMIT'] = ''
+    atexit.register(os.unsetenv, 'DCOS_IMAGE_COMMIT')
+
+import dcos_installer.config  # noqa
```

Comment:
Sure thing. I figured that I will one day soon break the circular dependency between dcos_installer and gen which breaks in all kinds of weird ways, so this would be one of the main tasks for that